openshift/template.yaml

```diff
@@ -366,6 +366,13 @@ objects:
                   value: ${GT_APP_PORT}
               image: "${REGISTRY_IMAGE}:${IMAGE_TAG}"
               name: worker
+              startupProbe:
```

Comment:
The startup probe is for probing the app inside the container. The touch command doesn't probe the app in this case.

This command should be in the readiness probe and the startup probe can be left out if DVO allows it and if we are happy with the liveness probe.

The following I think should be the sequence of probes if we want to use `/tmp` to probe the worker pod.

A. Using startup probes
1. Startup probe probes the app (e.g. `ps ax | grep 'myapp'`)
2. Readiness probe probes the container (e.g. `ls /tmp`). Note that this is not anymore necessary because startup probe already probes the container, but DVO complains.
3. Liveness probe probes the app (e.g. `ps ax | grep 'myapp')

B. Without using startup probes
1. Readiness probe probes the container (e.g. `ls /tmp`)
2. Liveness probe probes the app (e.g. `ps ax | grep 'myapp')
    
If we want to use `/tmp/healthy`, we need to use it accordingly with its name `healthy`. Is this referring to the health of the container or the app? If it's the former then this needs to be baked in the image, otherwise the app should generate this file. This avoids confusion in understanding the use cases of these probes.

Reply:
Fair enough. Thanks for reviewing. 
**My Idea is:**
- Startup Probe fires once only during the startup of a container. So, creating a sample file and checking the exit code would ensure that container is running and that the sample file would further act as a reference to `liveness` and `readiness`. 

- In liveness and readiness, successful checking of this file (`/tmp/healthz`) ensures at least one long-running process on the other side is active (Mostly Application process) that enables the container to stay alive.

Reply:
I understand how the startup probe is used here  However we should not use it to create a file for other probes as this becomes an unnecessary dependency and an error on creating the file can mean something else.

The question that startup probe is trying to answer is:

`Has the application started?`

The probe needs to answer this even if it can only provide a lesser accurate answer.

If you can guarantee that there is only one process inside the container and that is the application, then I think it is alright for now to simply even check a file/dir that already exists.

For example:

`ls /tmp`

We can even use this for both liveness and readiness, and startup probes. But we should put a `Todo` marker in it to enhance the liveness. Actually we should log a ticket for this and lqbel it as enhancement.

WDYT?

Reply:
Updated!